snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ecl/SnomedEclEvaluationRequest.java

```diff
@@ -691,6 +701,91 @@ private static Expression executeDescriptionSearch(BranchContext context, Expres
 				});
 	}
 	
+	protected Promise<Expression> eval(BranchContext context, final DialectAliasFilter dialectAliasFilter) {
+		final ListMultimap<String, String> languageMapping = SnomedDescriptionUtils.getLanguageMapping(context);
+		final Multimap<String, String> languageRefSetsByAcceptability = HashMultimap.create();
+		final ExpressionBuilder dialectQuery = Expressions.builder();
+		for (DialectAlias alias : dialectAliasFilter.getDialects()) {
+			final Set<String> acceptabilitiesToMatch = getAcceptabilityIds(alias.getAcceptability());
+			final Collection<String> languageReferenceSetIds = languageMapping.get(alias.getAlias());
+			
+			// empty acceptabilities or empty language reference set IDs mean that none of the provided values were valid so no match should be returned
+			if (acceptabilitiesToMatch.isEmpty() || languageReferenceSetIds.isEmpty()) {
+				return Promise.immediate(Expressions.matchNone());
+			}
+			
+			for (String acceptability : acceptabilitiesToMatch) {
+				languageRefSetsByAcceptability.putAll(acceptability, languageReferenceSetIds);
+			}
+		}
+		
+		languageRefSetsByAcceptability.asMap().forEach((key, values) -> {
+			Operator op = Operator.fromString(dialectAliasFilter.getOp());
+			switch (op) {
+			case EQUALS: 
+				dialectQuery.should(Expressions.matchAny(key + "In", values));
```

Comment:
A bit magical 🧙 The expression can break if either the keyword in the grammar or the document field names change.

Reply:
Yeah, I know, I wanted to spare a bunch of if statements and explicit calls to `acceptableIn`, `preferredIn` expression helper calls. 😅 Suggestions to improve this piece of code are welcome. 🙂